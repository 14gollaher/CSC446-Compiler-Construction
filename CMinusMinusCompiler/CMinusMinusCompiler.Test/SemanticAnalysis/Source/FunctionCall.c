int main()
{
	int y;
	y = y(/*WILL THIS BE OMMITED?*/ y ,  /*WILL THIS BE OMMITED?*/ y /*WILL THIS BE OMMITED?*/ , /*WILL THIS BE OMMITED?*/ 2 /*WILL THIS BE OMMITED?*/ )/*WILL THIS BE OMMITED?*/; /*SHOULD THIS BE ALLOWED?*/
	return -1;
}

int foo()
{
	int x;
	x = main/*WILL THIS BE OMMITED?*/();
	x = x(x, 2.3); /*SHOULD THIS BE ALLOWED?*/ 
	return -2;
}
/*This will be omitted!!!*/
