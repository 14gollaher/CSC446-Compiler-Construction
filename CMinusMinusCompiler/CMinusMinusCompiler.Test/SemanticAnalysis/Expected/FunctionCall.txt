** Symbol Table at Depth 2 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
y               Variable     2         -2         IntToken     2         -


** Symbol Table at Depth 2 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
x               Variable     2         -2         IntToken     2         -


** Symbol Table at Depth 1 **
Lexeme          Class        Depth     Offset    Type         Size      Value(r)
--------------------------------------------------------------------------------
main            Function     1         -         IntToken     2         -
foo             Function     1         -         IntToken     2         -


Completed processing FunctionCall.c